File: 15th April - Find Total Time Taken/15thApril.java
// Java solution
class Solution
{
    public static long totalTime(int n,int arr[],int time[])
    {
        long count=0;
        HashSet<Integer> hs=new HashSet<>();
        hs.add(arr[0]);
        for(int i=1;i<n;i++){
            if(hs.contains(arr[i])){
                count+=time[arr[i]-1];
            }else{
                hs.add(arr[i]);
                count+=1;
            }
        }
        return count;
    }
}

File: 15th April - Find Total Time Taken/15thApril.cpp
// C++ solution
class Solution {
  public:
    int totalTime(int n, vector<int> &arr, vector<int> &time) {
        // code here
        map<int,bool> visited;
        int t=-1;
        for( int i=0; i<n; i++ )
            visited[arr[i]] = false;
        for( int i=0; i<n; i++ ){
            if( !visited[arr[i]] ){
                visited[arr[i]] = true;
                t++;
            }
            else
                t+=time[arr[i]-1];
        }
        return t;
    }
};